From 086122bd263eac1d69b044e0400c4aad86f5751e Mon Sep 17 00:00:00 2001
Message-ID: <086122bd263eac1d69b044e0400c4aad86f5751e.1708471786.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1708471786.git.sharlatanus@gmail.com>
References: <cover.1708471786.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 27 Jan 2024 18:39:17 +0000
Subject: [PATCH 35/46] gnu: python-roman-datamodels: Speed up tests.

gnu/packages/astronomy.scm (python-roman-datamodels): Enable parallel
tests to speed them up.
[arguments] <#:test-flags>: Add it with "-n" option.
[native-inputs]: Add python-pytest-xdist.

Change-Id: I02b9e04a2f37fe05dc372c902b22d03da8f79605
---
 gnu/packages/astronomy.scm | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 7b8160a3ea..6a9c7e3183 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -4513,6 +4513,8 @@ (define-public python-roman-datamodels
     (build-system pyproject-build-system)
     (arguments
      (list
+      #:test-flags
+      #~(list "-n" "auto")
       #:phases
       #~(modify-phases %standard-phases
           (add-after 'unpack 'set-env
@@ -4529,6 +4531,7 @@ (define-public python-roman-datamodels
     (native-inputs (list python-pytest
                          python-pytest-doctestplus
                          python-pytest-env
+                         python-pytest-xdist
                          python-semantic-version
                          python-setuptools-scm))
     (home-page "https://github.com/spacetelescope/roman_datamodels")
-- 
2.41.0

